uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/test/shared/controllers/sub-controllers/toil-request-ctrl_test.js

```diff
@@ -0,0 +1,180 @@
+        $provide.value('LeaveRequestAPI', _LeaveRequestAPIMock_);
+        $provide.value('api.optionGroup', _OptionGroupAPIMock_);
+        $provide.value('PublicHolidayAPI', _PublicHolidayAPIMock_);
+      }));
+
+      beforeEach(inject(function (_$log_, _$controller_, _$rootScope_,
+        _AbsenceTypeAPI_, _TOILRequestInstance_) {
+
+        $log = _$log_;
+        $rootScope = _$rootScope_;
+        $controller = _$controller_;
+        AbsenceTypeAPI = _AbsenceTypeAPI_;
+        TOILRequestInstance = _TOILRequestInstance_;
+        modalInstanceSpy = jasmine.createSpyObj('modalInstanceSpy', ['dismiss', 'close']);
+
+        spyOn($log, 'debug');
+        spyOn(AbsenceTypeAPI, 'all').and.callThrough();
+        spyOn(AbsenceTypeAPI, 'calculateToilExpiryDate').and.callThrough();
+        spyOn(TOILRequestInstance, 'init').and.callThrough();
+      }));
+
+      describe('sick request', function () {
+        var parentRequestCtrl;
+
+        beforeEach(function () {
+          var directiveOptions = {
+            contactId: CRM.vars.leaveAndAbsences.contactId,
+            leaveType: 'sick'
+          };
+
+          initTestController(directiveOptions);
+          parentRequestCtrl = $controller('RequestCtrl')
+        });
+
+        it('is called', function () {
+          expect($log.debug).toHaveBeenCalled();
+        });
+
+        it('inherited from request controller', function(){
+          expect($ctrl instanceof parentRequestCtrl.constructor).toBe(true);
+        });
+
+        it('has leave type set to sick', function () {
+          expect($ctrl.isLeaveType('toil')).toBeTruthy();
+        });
+
+        it('calls init on sickness instance', function () {
+          expect(TOILRequestInstance.init).toHaveBeenCalledWith({
+            contact_id: CRM.vars.leaveAndAbsences.contactId
+          });
+        });
+
+        it('loads toil amounts', function () {
+          expect(Object.keys($ctrl.toilAmounts).length).toBeGreaterThan(0);
+        });
+
+        it('gets absence types with true sick param', function () {
+          expect(AbsenceTypeAPI.all).toHaveBeenCalledWith({
+            allow_accruals_request: true
+          })
+        });
+
+        it('cannot submit request', function () {
+          expect($ctrl.canSubmit()).toBe(false);
+        });
+
+        describe('with selected duration and dates', function () {
+          beforeEach(function () {
+            var toil_accrue = optionGroupMock.specificObject('hrleaveandabsences_toil_amounts', 'name', 'quarter_day');
+
+            setTestDates(date2016, date2016);
+            $ctrl.request.toilDurationHours = 1;
+            $ctrl.request.updateDuration();
+            $ctrl.request.toil_to_accrue = toil_accrue.value;
+          });
+
+          it('cannot submit request', function () {
+            expect($ctrl.canSubmit()).toBe(true);
+          });
+
+          it('sets expiry date', function() {
+            expect($ctrl.expiryDate).toEqual('2016-07-08');
```

Comment:
We should not use static values for checking equality. Is it coming from mockedData? Then we can use mockData variables here.